src/main/java/org/apache/commons/imaging/Imaging.java

```diff
@@ -857,20 +837,19 @@ public static Dimension getImageSize(final File file) throws ImageReadException,
      *
      * @param file
      *            File containing image data.
-     * @param params
-     *            Map of optional parameters, defined in ImagingConstants.
+     * @param params optional parameters.
      * @return The width and height of the image.
      * @throws ImageReadException if it fails to parse the image
      * @throws IOException if it fails to read the image data
      */
-    public static Dimension getImageSize(final File file, final Map<String, Object> params)
+    public static Dimension getImageSize(final File file, final ImagingParameters params)
```

Comment:
Not sure about this API. Callers need to know the type of the image being read in order to avoid a `ClassCastException`. Possible options:
1. Make it possible to "upgrade" the parameters object internally and copy the values into an instance of the correct type.
2. Remove this method (and others that accept `ImagingParameters`) altogether and only allow parameters to be passed using the methods on the parser. We could make the parsers more easily accessible by making the `getImageParser` method public.

Reply:
@darkma773r I went with option 2. It reduces the `Imaging.java` size, and prevents the god-class issue in  case we added more image formats later. IMHO, it will be simpler to either improve the `ImageParser`'s API, or think in how to better handle parameters in the `Imaging` class (like merging parameters, of providing factories as you suggested).

But for 1.0 this should be better than the old `Hashmap`, and avoid runtime exceptions. Thank you!!!